File: solutions/0088-merge-sorted-array/solution.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {

        int c = m+n-1;
        
        if(m==0){
            for(int i=0; i<n; i++){
                nums1[i] = nums2[i];
            }
        }
        
        else if(n==0){
            
        }
         else{
             int i=m-1;
             int j=n-1;
             
        while(c>=0 && i>=0 && j>=0 ){
            if(nums1[i]>=nums2[j]){
                nums1[c] = nums1[i];
                i--;
                c--;
            }
            else{
                nums1[c] = nums2[j];
                j--;
                c--;
            }
        }
             
             while(c>=0 && i>=0){
                 nums1[c] = nums1[i];
                 c--;
                 i--;
             }
             
             
             while(c>=0 && j>=0){
                 nums1[c] = nums2[j];
                 j--;
                 c--;
             }
             
         }
    
    }
}

File: solutions/0088-merge-sorted-array/solution.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int p = m+n-1;
        int end1 = m-1;
        int end2 = n-1;
        while(end2>=0 and end1>=0){
            if(nums1[end1]<=nums2[end2]){
                nums1[p]=nums2[end2];
                end2--;
            }
            else{
                nums1[p]=nums1[end1];
                end1--;
            }
            p--;
        }
        while(end2>=0){
            nums1[p]=nums2[end2];
            end2--;
            p--;
        }

    }
};
